python/chemiscope/explore.py

```diff
@@ -197,6 +198,121 @@ def soap_pca_featurize(frames, environments=None):
+        is ``"extensions/"``.
+
+    :param device: a torch device to use for the calculation. If ``None``, the function
+        will use the options in model's ``supported_device`` attribute.
+
+    :returns: a function that takes a list of frames and returns the features.
+
+    To use this function, additional dependencies are required, specifically,
+    `metatensor`_. It can be installed with the following command:
+
+    .. code:: bash
+
+        pip install chemiscope[metatensor]
+
+    Here is an example using a pre-trained `metatensor`_ model, stored as a
+    ``model.pt`` file with the compiled extensions stored in the ``extensions/``
+    directory. To obtain the details on how to get it, see metatensor `tutorial
+    <https://lab-cosmo.github.io/metatrain/latest/getting-started/usage.html>`_.
+    The frames are obtained by reading structures from a file that `ase <ase-io_>`_
+    can read.
+
+    .. code-block:: python
+
+        import chemiscope
+        import ase.io
+
+        # Read the structures from the dataset
+        frames = ase.io.read("data/explore_c-gap-20u.xyz", ":")
+
+        # Provide model file ("model.pt") to 'metatensor_featurizer', it's result will
+        # be visualised with a chemiscope widget
+        chemiscope.explore(
+            frames,
+            featurize=chemiscope.metatensor_featurizer("model.pt"),
+        )
+
+    For more examples, see the related `documentation <chemiscope-explore-metatensor>`_.
+
+    .. _metatensor: https://docs.metatensor.org/latest/index.html
+    .. _chemiscope-explore-torchscript:
+        https://chemiscope.org/docs/examples/7-explore-advanced.html#example-with-metatensor-model
+    .. _torch.jit.script:
+        https://pytorch.org/docs/stable/generated/torch.jit.script.html#torch.jit.script
+    """
+
+    # Check if dependencies were installed
+    try:
+        from metatensor.torch.atomistic import ModelOutput
+        from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+    except ImportError as e:
+        raise ImportError(
+            f"Required package not found: {e}. Please install the dependency using "
+            "'pip install chemiscope[metatensor]'."
+        )
+
+    # Initialize metatensor calculator
+    mtt_calculator = MetatensorCalculator(
+        model=model,
+        extensions_directory=extensions_directory,
+        check_consistency=True,
+        device=device,
+    )
+
+    def get_llf(model, frame):
+        """
+        Run the model on a single atomic structure and extract the features
+
+        :param model: a Metatensor calculator model
+        :param frame: an atomic structure to be featurized
+        """
+        # Run the model
+        opts = {"features": ModelOutput(per_atom=True)}
+        output = model.run_model(frame, opts)
+
+        # Extract and return features
+        llf = output["features"]
+        return llf.block().values.detach().cpu().numpy()
+
+    def featurize(frames):
```

Comment:
this should accept an `environements` parameter, which can then be translated to `selected_atoms` and given to the metatensor model